src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteImporter.kt

```diff
@@ -386,83 +376,167 @@ class PlantingSiteImporter(
+    // them didn't fall within the subzone).
+    val clustersInSubzones: List<List<MonitoringPlotsRow>> =
+        subzoneRows.flatMap { subzoneRow ->
+          var plotNumber = 0
+
+          val subzoneClusters =
+              allPlots
+                  .map { cluster -> cluster.filter { it.coveredBy(subzoneRow.boundary!!) } }
+                  .filter { it.isNotEmpty() }
+                  .map { cluster ->
+                    cluster.map { polygon ->
+                      plotNumber++
+
+                      MonitoringPlotsRow(
+                          boundary = polygon,
+                          createdBy = userId,
+                          createdTime = now,
+                          fullName = "${subzoneRow.fullName}-$plotNumber",
+                          modifiedBy = userId,
+                          modifiedTime = now,
+                          name = "$plotNumber",
+                          plantingSubzoneId = subzoneRow.id,
+                      )
+                    }
+                  }
+
+          log.debug("Generated $plotNumber plots for subzone ${subzoneRow.fullName}")
+
+          subzoneClusters
+        }
+
+    // Now we have a list of clusters of monitoring plots, where each cluster's plots are all in
+    // the same subzone. Add permanent monitoring cluster numbers in random order to all
+    // the 4-plot clusters.
+    val permanentClusters =
+        clustersInSubzones
+            .filter { it.size == 4 }
+            .shuffled()
+            .mapIndexed { clusterIndex, plotsRows ->
+              plotsRows.mapIndexed { subplotIndex, plotsRow ->
+                plotsRow.copy(
+                    permanentCluster = clusterIndex + 1,
+                    permanentClusterSubplot = subplotIndex + 1,
+                )
               }
+            }
 
-      log.debug("Generated ${plots.size} plots for subzone ${getFullSubzoneName(subzoneFeature)}")
+    val nonPermanentClusters = clustersInSubzones.filter { it.size != 4 }
 
-      getFullSubzoneName(subzoneFeature) to plots
-    }
+    return (permanentClusters + nonPermanentClusters).flatten()
   }
 
   /**
-   * Generates a list of all potential monitoring plots for an entire planting site.
+   * Generates a list of all potential monitoring plots for an entire planting site, clustered in
+   * 2x2 groups. Only plots that fall within the planting site are included.
    *
-   * This effectively divides the site into a grid of 25m squares and returns a polygon for each
-   * square.
+   * This effectively divides the site into a grid of 50m squares, divides each square into four 25m
+   * squares, and returns a polygon for each of the smaller squares.
    */
-  private fun generateAllPlots(siteFeature: ShapefileFeature): List<Polygon> {
-    val plots = mutableListOf<Polygon>()
+  private fun generatePlotBoundaries(siteFeature: ShapefileFeature): List<List<Polygon>> {
```

Comment:
just for discussion, any benefit to introducing clusters as a data construct here.. ?return a list of clusters vs list of list of polygons.

Reply:
Yeah, that might make this easier to follow. Good suggestion.